dspace-api/src/main/java/org/dspace/external/provider/impl/OpenAIREFundingDataProvider.java

```diff
@@ -0,0 +1,478 @@
+        }
+
+        /**
+         * Set the external data display name
+         * 
+         * @param displayName
+         * @return ExternalDataObjectBuilder
+         */
+        public ExternalDataObjectBuilder setDisplayValue(String displayName) {
+            this.externalDataObject.setDisplayValue(displayName);
+            return this;
+        }
+
+        /**
+         * Set the external data value
+         * 
+         * @param value
+         * @return ExternalDataObjectBuilder
+         */
+        public ExternalDataObjectBuilder setValue(String value) {
+            this.externalDataObject.setValue(value);
+            return this;
+        }
+
+        /**
+         * Set the external data id
+         * 
+         * @param id
+         * @return ExternalDataObjectBuilder
+         */
+        public ExternalDataObjectBuilder setId(String id) {
+            this.externalDataObject.setId(id);
+            return this;
+        }
+
+        /**
+         * Add metadata dc.identifier
+         * 
+         * @param metadata identifier
+         * @return ExternalDataObjectBuilder
+         */
+        public ExternalDataObjectBuilder addIdentifier(String identifier) {
+            this.externalDataObject.addMetadata(new MetadataValueDTO("dc", "identifier", null, null, identifier));
+            return this;
+        }
+
+        /**
+         * Add metadata project.funder.name
+         * 
+         * @param metadata funderName
+         * @return ExternalDataObjectBuilder
+         */
+        public ExternalDataObjectBuilder addFunderName(String funderName) {
+            this.externalDataObject.addMetadata(new MetadataValueDTO("project", "funder", "name", null, funderName));
+            return this;
+        }
+
+        /**
+         * Add metadata project.funder.identifier
+         * 
+         * @param metadata funderId
+         * @return ExternalDataObjectBuilder
+         */
+        public ExternalDataObjectBuilder addFunderID(String funderID) {
+            this.externalDataObject
+                    .addMetadata(new MetadataValueDTO("project", "funder", "identifier", null, funderID));
+            return this;
+        }
+
+        /**
+         * Add metadata dc.title
+         * 
+         * @param metadata awardTitle
+         * @return ExternalDataObjectBuilder
+         */
+        public ExternalDataObjectBuilder addAwardTitle(String awardTitle) {
+            this.externalDataObject.addMetadata(new MetadataValueDTO("dc", "title", null, null, awardTitle));
+            return this;
+        }
+
+        /**
+         * Add metadata oaire.fundingStream
+         * 
+         * @param metadata fundingStream
+         * @return ExternalDataObjectBuilder
+         */
+        public ExternalDataObjectBuilder addFundingStream(String fundingStream) {
+            this.externalDataObject
+                    .addMetadata(new MetadataValueDTO("oaire", "fundingStream", null, null, fundingStream));
```

Comment:
It appears this field doesn't exist in out of the box DSpace.  I'm getting this error `bad_dublin_core schema=oaire.fundingStream.null. Metadata field does not exist!`

I believe it's because the `openaire4-types.xml` file isn't listed in our `registry.metadata.load` setting in dspace.cfg.  Could you append it there as part of this PR? https://github.com/DSpace/DSpace/blob/main/dspace/config/dspace.cfg#L808-L820

Reply:
Yes, I added OpenAIRE types to DSpace.cfg. 
Perhaps we never did it because we wanted an optional feature for only for people that want to really use openaire. But as is (where) raises some avoidable setup issues.